### PR TITLE
Stable versioning for 9.0 GA

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Includes:
- Changes to enable stable versioning for 9.0 GA release
- Fix for native build required due to build host machine changes - a backport from main
- Suppression of SYSLIB0057 - deprecated API in 9.0 GA SDK

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Why the NetCurrent change in the managed lib?

**Comment by NikolaMilosavljevic** [comment]
> Why the NetCurrent change in the managed lib?

`NetMinimum` evaluates to `net8.0`. We want to target `net9.0`

**Comment by ViktorHofer** [comment]
Ok but why? Just curious.

**Comment by NikolaMilosavljevic** [comment]
> Ok but why? Just curious.

All previous GA releases of `dotnet-mage` have been targeting the matching version of .NET, so this just follows the established model.

**Comment by ViktorHofer** [comment]
We generally recommend to target the minimum supported .NET version in nuget packages when possible. I.e. all our libraries packages like System.Text.Json target NetMinimum (`net8.0`) in addition to NetCurrent (`net9.0` for 9.0). This allows the package to be used with all in-support runtimes (broader reach). `NetMinimum` automatically gets updated to `net9.0` when building from source. The Arcade SDK handles that. The decision is of course yours, just wanted to point that option out.

File: src/clickonce/MageCLI/MageCLI.csproj [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-    <TargetFramework>$(NetMinimum)</TargetFramework>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
     <AssemblyName>dotnet-mage</AssemblyName>
     <OutputType>Exe</OutputType>
     <SignAssemblyAttribute>true</SignAssemblyAttribute>
     <RCSuppressManifest>true</RCSuppressManifest>
     <AssemblyAttributeClsCompliant>false</AssemblyAttributeClsCompliant>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
+    <NoWarn>$(NoWarn);SYSLIB0057</NoWarn>
```
Comment (ViktorHofer) [inline src/clickonce/MageCLI/MageCLI.csproj:12]: Is it safe to ignore the deprecated API? You might want to add a comment here explaining what this is about, i.e. which API.

Comment (NikolaMilosavljevic) [reply src/clickonce/MageCLI/MageCLI.csproj:12]: I will add a comment. We will fix this in 10.0.

**Comment by ViktorHofer** [comment]
Targeting the minimum makes sense for tools as well as you still need a runtime installed to run the tool. By targeting the lowest supported runtime and adding a roll-forward switch, the tool will work on all platforms that we care about. Sure, just consider my suggestion. No need for any immediate action here.

**Comment by NikolaMilosavljevic** [comment]
> Targeting the minimum makes sense for tools as well as you still need a runtime installed to run the tool. By targeting the lowest supported runtime and adding a roll-forward switch, the tool will work on all platforms that we care about. Sure, just consider my suggestion. No need for any immediate action here.

That does make sense, thanks. Let me add the roll-forward switch and keep support for 8.0, i.e `<RollForward>Major</RollForward>`

**Pull request merged**